x-pack/docs/en/security/authentication/oidc-guide.asciidoc

```diff
@@ -0,0 +1,649 @@
+    _exactly_ the same as the one <<oidc-guide-op, configured with the OP upon registration>> and will
+    typically be +$\{kibana-url}/api/security/v1/oidc+ where _$\{kibana-url}_ is the base URL for your {kib} instance
+
+op.authorization_endpoint::
+    The URL for the Authorization Endpoint in the OP. This is where the user's browser
+    will be redirected to start the authentication process. The value for this setting should be provided by your
+    OpenID Connect Provider.
+
+op.token_endpoint::
+    The URL for the Token Endpoint in the OpenID Connect Provider. This is the endpoint where
+    {es} will send a request to exchange the code for an ID Token, in the case where the Authorization Code
+    flow is used. The value for this setting should be provided by your OpenID Connect Provider.
+
+op.userinfo_endpoint::
+    (Optional) The URL for the UserInfo Endpoint in the OpenID Connect Provider. This is the endpoint of the OP that
+    can be queried to get further user information, if required. The value for this setting should be provided by your
+     OpenID Connect Provider.
+
+op.endsession_endpoint::
+    (Optional) The URL to the End Session Endpoint in the OpenID Connect Provider. This is the endpoint where the user's
+    browser will be redirected after local logout, if the realm is configured for RP initiated Single Logout and
+    the OP supports it. The value for this setting should be provided by your OpenID Connect Provider.
+
+op.jwkset_path::
+    The path to a file containing a JSON Web Key Set with the key material that the OpenID Connect
+    Provider uses for signing tokens and claims responses. The path is resolved relative to the {es}
+    config directory.
+    {es} will automatically monitor this file for changes and will reload the configuration whenever
+    it is updated. The value for this setting should be provided by your OpenID Connect Provider.
+
+claims.principal:: See <<oidc-claims-mapping>>.
+claims.groups:: See <<oidc-claims-mapping>>.
+
+A final piece of configuration of the OpenID Connect realm is to set the `Client Secret` that was assigned
+to the RP during registration in the OP. This is a secure setting and as such is not defined in the realm
+configuration in `elasticsearch.yml` but added to the {ref}/secure-settings.html[elasticsearch keystore].
+For instance
+
+
+[source,sh]
+----
+bin/elasticsearch-keystore add xpack.security.authc.realms.oidc.oidc1.rp.client_secret
+----
+
+
+NOTE: According to the OpenID Connect specification, the OP should also make their configuration
+available at a well known URL, which is the concatenation of their `Issuer` value with the
+`.well-known/openid-configuration` string. For example: `https://op.org.com/.well-known/openid-configuration`
+That document should contain all the necessary information to configure the OpenID Connect realm in {es}.
+
+
+[[oidc-claims-mapping]]
+==== Claims mapping
+
+===== Claims and scopes
```

Comment:
Is this sub-heading required?  It's a little unusual to have an empty section followed by another heading.

Reply:
It felt better to differentiate between teh subsections under the same topic, I can see if a paragraph up top makes sense